site/src/App.tsx

```diff
@@ -34,6 +34,18 @@ export default function App() {
     usingSystemTheme ? isSystemDark() : localStorage.getItem('theme') === 'dark',
   );
   const [cookies] = useCookies(['user']);
+  const [prevDarkMode, setPrevDarkMode] = useState(false); // light theme is default on page load
+
+  /**
+   * we run this check at render-time and compare with previous state because a useEffect
+   * would cause a flicker for dark mode users on page load since the first render would be without
+   * the data-theme property set (light would be used by default)
+   */
+  if (darkMode != prevDarkMode) {
+    // Theme styling is controlled by data-theme attribute on body being set to light or dark
+    document.querySelector('body')!.setAttribute('data-theme', darkMode ? 'dark' : 'light');
```

Comment:
Is there a reason we don't just say `document.body`?
```suggestion
    document.body.setAttribute('data-theme', darkMode ? 'dark' : 'light');
```